lib/prime.rb

```diff
@@ -32,15 +32,51 @@ def prime_division(generator = Prime::Generator23.new)
 
   # Returns true if +self+ is a prime number, else returns false.
   def prime?
-    return self >= 2 if self <= 3
-    return true if self == 5
-    return false unless 30.gcd(self) == 1
-    (7..Integer.sqrt(self)).step(30) do |p|
-      return false if
-        self%(p)    == 0 || self%(p+4)  == 0 || self%(p+6)  == 0 || self%(p+10) == 0 ||
-        self%(p+12) == 0 || self%(p+16) == 0 || self%(p+22) == 0 || self%(p+24) == 0
-    end
-    true
+    # https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test
+    # https://arxiv.org/pdf/1509.00864.pdf
+    #
+    # if n < 3,317,044,064,679,887,385,961,981,
+    # it is enough to test a = 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, and 41.
+    # and the result will be deterministic.
+    #
+    return false if self < 2
+
+    if self > 3317044064679887385961981
+      raise ArgumentError, ".prime? is only valid for values less than 3,317,044,064,679,887,385,961,981"
+    end
+    
+    if self < 6
+      return [false, false, true, true, false, true, false][self]
```

Comment:
I believe you have an extra ``false`` at the end of this Array, it should cover the 0..5 range.
The 6 is already covered by the even test ``self % 2 == 0``, leaving ``return [false, false, true, true, false, true][self]``.

Since 0 and 1 are already covered by the previous ``self < 2`` test and only 4 is false in this range, you could use ``return self != 4 if self < 6`` and remove the Array completely.

Reply:
Thanks for your feedback, I've updated this section to:

```
    return false if self < 2
    return true  if self < 4
    return false if self.even?
```